Package build fails without a running syslog daemon

I noticed that when building pn with Docker, the tests of the vendored syslog package are failing (and with that the package build).

```
make[1]: Entering directory `/data/git-debian-0.12'
mkdir -p /data/git-debian-0.12/debian/go_build/src/github.com/Jimdo/
rsync -a --cvs-exclude --exclude=debian --exclude=git-hooks /data/git-debian-0.12/. /data/git-debian-0.12/debian/go_build/src/github.com/Jimdo/periodicnoise
go get `go list -f '{{range .Deps}}{{print .}} {{end}}' github.com/Jimdo/periodicnoise` github.com/Jimdo/periodicnoise
go get `go list -f '{{range .Deps}}{{print .}} {{end}}' github.com/Jimdo/periodicnoise/cmd/pn` github.com/Jimdo/periodicnoise/cmd/pn
make[1]: Leaving directory `/data/git-debian-0.12'
   # Skipping dh_auto_build - empty override
   debian/rules override_dh_auto_test
make[1]: Entering directory `/data/git-debian-0.12'
cd /data/git-debian-0.12/debian/go_build/src/github.com/Jimdo/periodicnoise && go test ./...
?   	github.com/Jimdo/periodicnoise	[no test files]
ok  	github.com/Jimdo/periodicnoise/cmd/pn	0.249s
--- FAIL: TestNew (0.00 seconds)
	syslog_test.go:249: New() failed: Unix syslog delivery error
--- FAIL: TestNewLogger (0.00 seconds)
	syslog_test.go:261: Unix syslog delivery error
--- FAIL: TestDial (0.00 seconds)
	syslog_test.go:279: Dial() failed: Unix syslog delivery error
FAIL
FAIL	github.com/Jimdo/periodicnoise/syslog	4.082s
make[1]: *** [override_dh_auto_test] Error 1
make[1]: Leaving directory `/data/git-debian-0.12'
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
debuild: fatal error at line 1325:
dpkg-buildpackage -rfakeroot -D -us -uc -i.git -I.git -b failed
debuild -i\.git -I.git -uc -us -b returned 29
Couldn't run 'debuild -i\.git -I.git -uc -us -b'
```

Something like `DEB_BUILD_OPTIONS=nocheck` didn't help either.

There are several ways to fix this problem (and probably more):

* Support `DEB_BUILD_OPTIONS=nocheck` to skip tests.
* Only run our tests, ignoring the syslog package.
* Add syslog to build dependencies (although this doesn't ensure it's actually running).
* Change tests to check for presence of syslog first.

--- a/debian/rules
+++ b/debian/rules
@@ -26,7 +26,7 @@ override_dh_auto_configure:
 	go get `go list -f '{{range .Deps}}{{print .}} {{end}}' $(GO_COMMANDS)` $(GO_COMMANDS)
 
 override_dh_auto_test:
-	cd $(GOPATH)/src/$(GO_PACKAGE) && go test ./...
+	cd $(GOPATH)/src/$(GO_PACKAGE) && go test -short ./...
 
 override_dh_install:
 	dh_install --sourcedir=$(GOPATH)/bin pn $(DESTDIR)/usr/bin

